modules/vaos/app/services/eps/draft_appointment_service.rb

```diff
@@ -0,0 +1,236 @@
+# frozen_string_literal: true
+
+require_relative 'draft_appointment_service/service_error'
+
+module Eps
+  ##
+  # Service responsible for creating draft appointments based on referrals
+  # and gathering necessary related data like providers, slots and drive times.
+  #
+  class DraftAppointmentService
+    ##
+    # Initialize the service with the current user
+    #
+    # @param user [User] The current user
+    #
+    def initialize(user)
+      @user = user
+      @appointments_service = VAOS::V2::AppointmentsService.new(user)
+      @eps_provider_service = Eps::ProviderService.new(user)
+      @eps_appointment_service = Eps::AppointmentService.new(user)
+    end
+
+    ##
+    # Creates a draft appointment and gathers related information
+    #
+    # @param referral_id [String] The referral identifier
+    # @param pagination_params [Hash] Parameters for pagination when fetching appointments
+    # @return [Hash] Response containing draft appointment data or error information
+    #
+    def call(referral_id, pagination_params)
```

Comment:
Why not call this create_draft os something like it, like on the old code? The name makes it hard to understand what it does.

Reply:
@carlosfelix2 this follows a more idiomatic naming convention in which the name of the class identifies what it does, wherein the method call signifies the class itself taking care of it's intended action. I don't feel super strongly about this though and am fine with changing it